ext/winevt/winevt_channel.c

```diff
@@ -254,8 +264,23 @@ rb_winevt_channel_each(VALUE self)
     }
 
     status = is_subscribable_channel_p(hChannelConfig, winevtChannel->force_enumerate);
-    if (status != ERROR_SUCCESS)
+    EvtClose(hChannelConfig);
+
+    if (status == ERROR_OUTOFMEMORY) {
+      free(buffer);
+      buffer = NULL;
+      bufferSize = 0;
+
```

Comment:
Should close `winevtChannel->channels`.

Reply:
Fixed in https://github.com/fluent-plugins-nursery/winevt_c/pull/23/commits/4c55c6fa775a9f981114f84fd1fc8328ea62dfcc